src/sync.rs

```diff
@@ -37,6 +37,11 @@ impl<T> GILProtected<T> {
     pub fn get<'py>(&'py self, _py: Python<'py>) -> &'py T {
         &self.value
     }
+
+    /// Gain access to the inner value by giving proof that garbage collection is happening.
+    pub fn get_visit<'py>(&'py self, _visit: PyVisit<'py>) -> &'py T {
```

Comment:
I am on board with the concept but would suggest `traverse` as the method name.

Reply:
I think `traverse` is an excellent name here 👍